src/Lorentz/Contracts/TZBTC/V1/Impl.hs

```diff
@@ -59,32 +59,32 @@ type StorageC store = StorageContains store
 -- value from the storage as is.
 getSingleField
   :: forall store label val.
-    (StoreHasField store label val, NiceParameter val)
+    (StoreHasField store label val, NiceParameter val, Dupable store)
   => Label label -> Markdown -> Entrypoint (View () val) store
 getSingleField label entrypointDocDesc = do
   doc $ DDescription $ "This view returns " <> entrypointDocDesc <> "."
   view_ $ do drop @(); stToField $ fieldNameFromLabel label
 
 getTotalMinted
-  :: forall store. StorageC store => Entrypoint (View () Natural) store
+  :: forall store. (StorageC store, Dupable store) => Entrypoint (View () Natural) store
```

Comment:
It looks like you could add `Dupable store` to the `StorageC store` type itself (both here and in `V2`) and it would make all these other updates unnecessary.

Aside from a cleaner diff, I think it makes sense logically as well (and IIRC the same was done in other projects as well).